### ISSUE TITLE
Conflict between training and flashtorch visualizer

### ISSUE THREAD

**Issue by PaulCCCCCCH** [HaohanWang/Robustar_implementation#76]
There has been a weird bug that makes "start training" fail non-deterministically. It turns out to be caused by the flashtorch visualizer. In `back-end\modules\visualize_module\flashtorch_\saliency\backprop.py`, it registers some hooks which get triggered when models run `forward` and `backward` functions, which will break model training. This appears to be non-deterministic because training will work if you start Robustar and start training right away before you visualize anything.

We **cannot** have multiple copies of a model, because we should have a single source of truth for one Robustar instance (so that we can iteratively see how model predicts after training for a few epochs, etc.) 
Thus, a better solution is to **unregister** the hooks when we train a model.

[Reference for unregistering a hook](https://github.com/pytorch/pytorch/issues/5037)

**Comment by PaulCCCCCCH** [comment]
Fixed in #78 

After we produce model visualization images, we will unregister the handlers immediately. 

However, it does not prevent a user to start training first, then generate visualization. It does not give errors, but may affect the gradient for training.
